app/controllers/confirmation/ConfirmationController.scala

```diff
@@ -29,19 +33,43 @@ import scala.concurrent.ExecutionContext
 
 @Singleton
 class ConfirmationController @Inject()(
-                                        override val messagesApi: MessagesApi,
+                                        override implicit val messagesApi: MessagesApi,
                                         actions: Actions,
                                         val controllerComponents: MessagesControllerComponents,
                                         confirmationView: ConfirmationView,
-                                        config: FrontendAppConfig
+                                        config: FrontendAppConfig,
+                                        estatesConnector: EstatesConnector
                                       )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
 
+  private def personalRepName(personalRepresentative: PersonalRepresentativeType)
+                             (implicit request: Request[_]): String = {
+    personalRepresentative match {
+      case PersonalRepresentativeType(Some(EstatePerRepIndType(name, _, _, _, _, _, _, _)), None) =>
+        s"${name.firstName} ${name.lastName}"
+      case PersonalRepresentativeType(None, Some(EstatePerRepOrgType(name, _, _, _, _, _, _))) =>
+        name
+      case _ =>
+        val lang = request.lang
+        messagesApi("confirmationPage.personalRepresentative.default")(lang)
```

Comment:
I have a question about this sort of thing actually - is it good practice to declare vals for everything or is something like this a waste of memory?

Reply:
It's a waste of memory but I'm unsure when garbage collection would occur on the values.

Reply:
@jackheslop96 https://stackoverflow.com/questions/4138200/garbage-collection-on-a-local-variable